### DISCUSSION TITLE
Show GPX Track from File -  Show waypoint information from recorded gpx

### DISCUSSION THREAD

**Discussion by thofrey** [Helium314/SCEE#592] [Ideas]
Usecase
I use the app gpslogger (F-Droid) to collect some simple data in car, or on my bicycle. Gpslogger has a configurable annotation view with 9 buttons, where the meaning can be edited. For example: "Start Living Zone", "sign 50", "building - house". 

When a button is presssed, the value of the button is written to a gpx file. The gpx file contains waypoints (wpt) and the track (trk). The position of the wpt are used in the track. Only the wpt contain the information of the pressed button. 

Example
[gpslogger_test-01.gpx.txt](https://github.com/user-attachments/files/16486993/gpslogger_test-01.gpx.txt)

Here: "Zone 30 Start Ende" is the value in tag "name".

`<wpt lat="48.8904072" lon="8.235780083333333"><ele>182.867</ele><time>2024-08-02T08:59:15.000Z</time><name>Zone 30 Start Ende</name><src>gps</src></wpt>`

`<trkpt lat="48.8904072" lon="8.235780083333333"><ele>182.867</ele><time>2024-08-02T08:59:15.000Z</time><extensions><gpxtpx:TrackPointExtension><gpxtpx:bearing>197.399</gpxtpx:bearing><gpxtpx:speed>1.0730728</gpxtpx:speed></gpxtpx:TrackPointExtension></extensions><geoidheight>0</geoidheight><src>gps</src><sat>9</sat><hdop>1.227</hdop><vdop>1.934</vdop><pdop>2.291</pdop></trkpt>`

Current behavior in SCEE
The track is imported. The waypoint information is not available.

My workaround
I (must) use ID-Editor, just pull the gpx in the map and I can see track, waypoints and information of the waypoints. Works fine, but requires for me a lot more motivation. 

My idea for discussion

- It would be nice, the wpt information can be shown in SCEE, so I can map the gpx information promptly.
- Both, trk- and wpt-information should be used from the file (to have line and points)
- From wpt the name and/or description might be relevant in this order.
- The information could be displayed as label. In this case an additional flag in the gpx-import dialogue to show (yes|no) might be useful.
- As alternative it could also be fine to make the wpt clickable and then show the information.

Hmm?

**Comment by Helium314** [comment]
Just showing name or description for a point is _relatively_ little work (still an hour or two with testing), but the more we get into a niche of a niche (of a niche), the less I'm inclined to spend a lot of time. PR can be done of course.

**Reply by thofrey** [reply]
As there were no other comments, it is obviously nothing a lot of users would use. Please use your time for the things you consider as more important. Many thanks for your answer.

**Comment by Helium314** [comment]
Acually MapLibre supports setting GeoJson from string. So when you convert the GPX to GeoJson, all that needs to be implemented is the display style.

**Reply by thofrey** [reply]
Wow, thanks for your feedback. If you think this is not that much work to upload GeoJson, to change the display style - and it would be a useful(!) feature - I would be happy. Have a happy new year.

**Reply by mnalis** [reply]
- ref: https://github.com/Helium314/SCEE/pull/706

**Reply by thofrey** [reply]
Just installed 60.1 and it works perfect. It is really cool and I can test some more and update my script to GeoJson. Many thanks.

![Screenshot_20250114_052206](https://github.com/user-attachments/assets/14e532be-7397-4b52-aa98-6ac7e7735c3a)
